File: esports-backend/src/main/java/com/esports/controller/UserController.java
package com.esports.controller;

import com.esports.dto.ApiResponse;
import com.esports.dto.UserProfileUpdateDTO;
import com.esports.dto.UserResponseDTO;
import com.esports.service.UserService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.NoSuchElementException;

@RestController
@RequestMapping("/api/users")
@CrossOrigin(origins = "http://localhost:5173")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/me")
    public ResponseEntity<?> getCurrentUser(Authentication authentication) {
        try {
            UserResponseDTO user = userService.getCurrentUser(authentication.getName());
            return ResponseEntity.ok(user);
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to get user data"));
        }
    }
    
    @PutMapping("/me")
    public ResponseEntity<?> updateProfile(@Valid @RequestBody UserProfileUpdateDTO profileData, Authentication authentication) {
        try {
            UserResponseDTO updatedUser = userService.updateUserProfile(authentication.getName(), profileData);
            return ResponseEntity.ok(updatedUser);
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to update profile"));
        }
    }
}

File: esports-backend-dotnet/Upkart1/Controllers/UserController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using System.ComponentModel.DataAnnotations;
using System.Security.Claims;
using Upkart1.DTO;
using Upkart1.Services;

namespace Upkart1.Controllers
{
    [ApiController]
    [Route("api/users")]
    public class UserController : ControllerBase
    {
        private readonly IUserService _userService;

        public UserController(IUserService userService)
        {
            _userService = userService;
        }

        [HttpGet("me")]
        public IActionResult GetCurrentUser()
        {
            try
            {
                var email = User.FindFirst(ClaimTypes.Email)?.Value;
                var user = _userService.GetCurrentUser(email);
                return Ok(user);
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<object>.ErrorResponse("User not found"));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<object>.ErrorResponse("Failed to get user data"));
            }
        }

        [HttpPut("me")]
        public IActionResult UpdateProfile([FromBody] UserProfileUpdateDTO profileData)
        {
            try
            {
                var email = User.FindFirst(ClaimTypes.Email)?.Value;
                var updatedUser = _userService.UpdateUserProfile(email, profileData);
                return Ok(updatedUser);
            }
            catch (Upkart1.Exceptions.ValidationException ex)
            {
                return BadRequest(ApiResponse<object>.ErrorResponse("Validation failed", ex.ValidationErrors));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<object>.ErrorResponse("User not found"));
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Profile update error: {ex.Message}");
                return StatusCode(500, ApiResponse<object>.ErrorResponse("Failed to update profile"));
            }
        }
    }
}